src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/Utilities/Checksum.Builder.cs

```diff
@@ -31,131 +48,161 @@ private enum TypeKind : byte
         [ThreadStatic]
         private static byte[]? s_buffer;
 
+#if NET5_0_OR_GREATER
         private readonly IncrementalHash _hash;
+#else
+        private readonly SHA256 _hash;
+#endif
 
         public Builder()
         {
-            _hash = IncrementalHashPool.Default.Get();
+            _hash = s_incrementalHashPool.Get();
+
+#if !NET5_0_OR_GREATER
```

Comment:
Is there a reason we use `#if !NET_5_0_OR_GREATER` in some places but `#if NET` in others? Would expect one type of condition. Or is there some specific API here that I'm missing in my review?

Reply:
Just a mistake on my part. I started with NET5_0_OR_GREATER, but then I realized that all of our .NET Core TFMs were already great than NET5_0_OR_GREATER and started to change them. Honestly, they should probably all be NET5_0_OR_GREATER